modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/persistence_impl.ftl

```diff
@@ -371,6 +371,16 @@ public class ${entity.name}PersistenceImpl extends BasePersistenceImpl<${entity.
 	 */
 	@Override
 	public void cacheResult(List<${entity.name}> ${entity.pluralVariableName}) {
+		<#if serviceBuilder.isVersionGTE_7_0_0()>
+			if ((valueObjectFinderCacheListThreshold >= 0) &&
+				(${entity.pluralVariableName}.size() > valueObjectFinderCacheListThreshold)) {
+
+				clearCache(${entity.pluralVariableName});
```

Comment:
@dantewang You do not need to clear entity cache. The reason we clear finder cache as a different result may be stored in cache. But for entity cache, the entity is not changed as all the cacheResult() happens after a fetch which means no update to the entities.